swift3/iam.py

```diff
@@ -373,6 +375,7 @@ def __call__(self, env, start_response):
         # Put the rules callback in the request environment so middlewares
         # further in the pipeline can call it when needed.
         env[IAM_RULES_CALLBACK] = self.rules_callback
+        env[IAM_RULES_CALLBACK].logger = self.logger
```

Comment:
We can't do this with a method.
On the other hand, the `matcher` has a logger that we could use.